addons/website_forum/models/forum.py

```diff
@@ -364,16 +405,12 @@ def _is_self_reply(self):
         for post in self:
             post.self_reply = post.parent_id.create_uid.id == post._uid
 
-    @api.depends('child_ids.create_uid', 'website_message_ids')
+    @api.depends('child_ids.create_uid')
     def _get_child_count(self):
-        def process(node):
-            total = len(node.website_message_ids) + len(node.child_ids)
-            for child in node.child_ids:
-                total += process(child)
-            return total
-
+        requests_data = self.env['forum.post'].read_group([('parent_id', 'in', self._ids)], ['parent_id'], ['parent_id'])
+        requests_mapped_data = dict((data['parent_id'][0], data['parent_id_count']) for data in requests_data)
         for post in self:
-            post.child_count = process(post)
+            post.child_count = requests_mapped_data.get(post.id, 0)
```

Comment:
I don't see the spec where we stop to consider a comment as a child?
It is the expected behavior?

Reply:
We discuss it with @sbuhl and, like in Stackoverflow, we doesn't want a comment to be consider as an answer. So, the elements in `website_message_ids` are no longer recorded in the `child_count` field. (The one used to show the number of answers).